### DISCUSSION TITLE
[New device support]: Yale "YMZB-100" Zigbee module made by Assa Abloy for Yale Doorman Classic, Yale Doorman V2N, Yale Doorman L3 and Yale Conexis L1, Yale Conexis L2, (and possibly other Yale locks for the European market)

### DISCUSSION THREAD

**Discussion by Hedda** [Koenkk/zigbee2mqtt#23915] [Devices]
Note! This is a new informal spiritual follow-up device feature request to https://github.com/Koenkk/zigbee2mqtt/issues/4778 and https://github.com/Koenkk/zigbee-herdsman-converters/issues/3326 (by @danieledwardgeorgehitchcock) as it seems like the old but already supported [Control4 ZYA-C4-MOD-S (previous known working Zigbee module for Yale Doorman V2/V2N or Yale Doorman L3 and Yale Conexis L1 / Yale Conexis L2)](https://www.zigbee2mqtt.io/devices/ZYA-C4-MOD-S.html) looks to be discontinued (or changed model number, and Control4's newer [AYR202-ZB-C4 module from Control4 looks to only be compatible with Yale Assure Lock and Yale nexTouch locks which use a different format](https://www.snapav.com/shop/en/snapav/yale-ayr202-zb-c4)), thus looking for other Zigbee modules specifically for Yale locks for the Yale oorman series and Yale Conexis series in Europe.

Update: Initial partial device support now available but missing major feature like lock and unlock:

* https://www.zigbee2mqtt.io/devices/ZYA-C4-MOD-S.html

Wondering if anyone has bought and tested **Assa Abloy's "YMZB-100" Zigbee module for Yale's European locks** as sounds as it should be compatible with at least Yale Doorman V2, Yale Doorman V2N, Yale Doorman L3, Yale Doorman L3S, Yale Conexis L1, and Yale Conexis L2 smart locks?

At least when searching there seems to be a Yale Doorman Zigbee module that has the Model/SKU/Article/Part-number "**_YMZB-100_**" which has been sold for the last couple of years in Norway that says to be manufactured by ASSA ABLOY, and as you may know; ASSA ABLOY is the parent company of Yale Europe, so might perhaps that is also be compatible with other Yale Doorman series and Yale Conexis series branded locks sold for the European market? 

- https://support.futurehome.no/hc/en-no/articles/360021021057-Yale-Doorman-Zigbee-Module

The product itself looks to be primarly marketed for the "Futurehome" and "Eva" smart home hubs (both Norwegian) that market themself as being compatible with most Zigbee devices:

* https://www.intin.no/produkt/futurehome-zigbee-modul-for-yale-doorman-v2n/

Though can now mostely be found pre-owned/used:

* https://www.finn.no/bap/forsale/ad.html?finnkode=367431254
* https://allematpriser.no/tilbud/elimport:product:4509475/
  * https://allematpriser.no/tilbud/elimport%3Aproduct%3A94350/
  * https://allematpriser.no/tilbud/elimport%3Aproduct%3A94329/

![image](https://github.com/user-attachments/assets/bfcc6cd2-7c10-4b56-86ff-ca06dde0d00e)

![image](https://github.com/user-attachments/assets/0e1347f2-659e-4836-b506-600cb53d2b6c)

Note! I have not yet bought it but know that Yale have previously used a different color for each different protocol and manufacturer and FYI, from pictures it looks like the color of this YMZB-100 Zigbee module looks to be red (or red-brown / wine-red / burgundy colored):

For reference:

Futurehome is a company in Norway making their own multi-protocol hub (Futurehome Smarthub 1v1-EU and Futurehome Smarthub II / 2.0) that is listed as being compatible with that "YMZB-100" Yale Doorman Zigbee Module:

* https://support.futurehome.no/hc/en-no/articles/360021021057-Yale-Doorman-Zigbee-Module

Eva (another smart home company from Norway) who also have their own Zigbee gateway looks to support the same module:

* https://evasmart.no/produkter/trygghet/modul-yale-doorman-dorlasmodul
  * https://hjelp.evasmart.no/hc/en-us/articles/6327009984657-Yale-Doorman-V2N-Zigbee-module

![image](https://github.com/user-attachments/assets/d36787f2-e75d-4039-b6e9-8ddf51b256c8)

![image](https://github.com/user-attachments/assets/8ddf784b-4ccd-4ff7-b696-a3a009fcb151)

PS: I believe that the Yale Doorman V2 and Yale Doorman V2N (and their successor Yale Doorman L3 and Yale Doorman L3S) are the most sold smart locks in Scandinavia if not the whole European Union, and similarly the Yale Conexis L1 / Yale Conexis L2 are the sold smart locks in the United Kingdom. And again, the Zigbee radio modules for those are not compatible with Yale Assure Lock or Yale nexTouch locks as those use a different format.

**Comment by ristoavila** [comment]
I'm also looking for this but havent found any one who is shipping it to Finland

**Comment by jpsmith16** [comment]
Commenting to follow, as I'd be interested in this module for a Conexis L1 with Zigbee2MQTT. Would also need to find someone that posts them to UK / Ireland!

**Reply by danieledwardgeorgehitchcock** [reply]
You'd probably be best getting the Control4 module - I got mine from a dealer on the UK C4 forums and did all of my converter development on a Conexis L1 - I've also tested it as compatible with the Conexis L2

**Reply by jpsmith16** [reply]
Thanks. I’ve not tried as have seen others post on C4 forums, but they were from a few years ago. So assumed it’d be a lost cause and went Z Wave route. But can only ask I suppose!

**Comment by cbjornerud** [comment]
I tried to add this device to my Yale Doorman Classic today. It was found, but not recognized by Home Assistant. I tested with both ZHA and Z2M. ZHA with a router close, Z2M with the coordinator itself very close.
I have read somewhere that this device is made only to work with systems approved by Yale. In this case EVA and Futurehome. I will probably go back to using the Verisure module and sell this to someone with EVA or Futurehome.
![IMG_0296](https://github.com/user-attachments/assets/72cbacb6-9e71-4f34-9e2f-50ebe1f15c02)

**Reply by Hedda** [reply]
It can probably be made to work by a developer but Zigbee2MQTT first needs a zigbee-herdsman converter:

https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

That translate the device customizations, as ZHA would need a ZHA Device Handler (a.k.a a custom quirk):

https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices

You can submit feature requests however you news to help by provising diagnostics logs and info.

**Comment by danieledwardgeorgehitchcock** [comment]
When creating the converter for the C4 module, I added a lot of the cluster attributes for the Doorman - the functionality between the Yale Keyfree and Doorman are very different (with different clusters) and the way this is handled by Control4 is to check the firmware attributes of the lock when adding the device to the Zigbee network, then reconfiguring the driver.

I'd assume that Yale wouldn't re-invent the wheel on core functional concepts of the modules created for different providers so, it is entirely feasible that a majority of the code I created for the C4 module will work with the Futurehome one.

That being said, I didn't add a lot of the Doorman functionality to my converter as I didn't have a YMZB-100 module or a Doorman lock to test and was therefore relying on feedback from others, which proved prohibitive.

I am happy to assist in whatever way I can to get this module integrated in Z2M though.. I don't think it'd take too much to get basic functionality working of the lock.

The question I kept asking myself while creating the C4 converter was, "yes I can do it, but what advantage is it bringing to the table other than having all my IoT devices on a single network?!" - these Zigbee modules are getting quite long in the tooth now, and as such, are missing a lot of the conveniences modern Yale modules provide.. The converter was a labour of love and curiosity for me - in the end, I reverted back to the Yale WiFi module with its solid integration with Home Assistant..

**Reply by Hedda** [reply]
Yale has a big lock company behind it (Assa Abloy) so they will at least making new versions of the Yale Doorman locks, at at least their official modules have been backwards compatible.

**Comment by danieledwardgeorgehitchcock** [comment]
If somebody can get this module to pair with Z2M and get the fingerprint from the database, I can at least begin to look into whether the work that I did for the C4 module can be ported.

Better still, if someone has a Doorman with a YMZB-100 module that they could send (lend / donate / sell (cheap)) to me in the UK, I should be able to get this integrated using the knowledge that I gained from integrating the C4 module..

**Reply by Pheomatar** [reply]
I have it. I connected it to my zigbee2mqtt but it says it is not supported. Can you tell me how can I send you the fingerprint or where do I find it? I would like to make this work because I have also a wireless module and I would not want to rely on the cloud for my lock to work.

**Reply by tm07x** [reply]
PM me if you need a module.

**Comment by Pheomatar** [comment]
Hi all,

I have a Yale Doorman V2N with the EVA Zigbee module and I’m trying to get it working locally with Zigbee2MQTT (and Home Assistant). I want to avoid using the cloud for my lock.

Here’s everything I’ve tried so far:

Pairing and Interview
The module pairs with Zigbee2MQTT and the interview completes successfully.
However, only linkquality is exposed—no lock, battery, PIN code, or other features.
The device does not report a modelID or manufacturerName in the logs during or after interview.
The logs just show:
Interviewing '0x680ae2fffe6baac8'
Successfully interviewed '0x680ae2fffe6baac8'
and then only linkquality is available in the state.
Custom Converter Attempts
I created a custom converter (yale_doorman.js) and tried:
Matching by IEEE address (fingerprint: [{ieeeAddr: ‘0x680ae2fffe6baac8’}])
Using a generic fingerprint (by type, endpoint, clusters)
Removing the fingerprint entirely (so the converter would match any device)
I included all relevant fromZigbee/toZigbee converters and exposes (lock, battery, pin code, etc.).
No matter what, only linkquality is exposed.
Other Troubleshooting
I tried waking up the lock (physically interacting with it) before and after pairing.
I tried re-pairing multiple times and restarting Zigbee2MQTT.
I checked the logs for any mention of modelID, manufacturerName, or clusters—nothing is reported.
I tried pairing with ZHA (Home Assistant’s Zigbee integration), but the result was the same: the lock is either not recognized or only exposes linkquality.
I checked for firmware updates for the lock/module—none available.
I found that the module works with the Futurehome hub (as described here), but I do not want to buy a new hub just for this.
What I See Under Exposes (Zigbee2MQTT)
Battery: Remaining battery in % (N/A%)
State: State of the lock
Lock state: Actual state of the lock (N/A)
Pin code: User, User type, User enabled, PIN code
Action source name: Source of the triggered action on the lock (N/A)
Action user: ID of user that triggered the action on the lock (N/A)
Auto relock time: The number of seconds to wait after unlocking a lock before it automatically locks again. 0=disabled
Sound volume: Sound volume of the lock
Action: Triggered action (e.g. a button click) (N/A)
Linkquality: Link quality (signal strength) (e.g. 102lqi)
Willing to Provide a Zigbee Sniffer Log
I am willing to buy a CC2531 USB stick from AliExpress and provide a Zigbee sniffer log if that would help with debugging or adding support.
Please let me know if you need a specific capture (e.g., during pairing/interview) and I will follow your instructions.
What I’m Looking For
Is there any way to get more debug information or a fingerprint for this device?
Would using a Zigbee sniffer help, or is there a way to enable more verbose logging in Zigbee2MQTT to capture the device’s clusters or attributes?
Has anyone managed to get this module working with Zigbee2MQTT or ZHA, or is there a custom handler/converter available?
Any advice or next steps would be greatly appreciated!
Thank you for your help!

**Comment by tehEyeCon** [comment]
i have made this converter and got it paired. it only gives me lockstate (locked/unlocked). 

Have not figured out the unlock/lock commands yet. I think it will require a pin. maybe someone better than me can figure it out.
```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const utils = require('zigbee-herdsman-converters/lib/utils');

const e = exposes.presets;

const tzAssaMastercode = {
    key: ['mastercode'],
    convertSet: async (entity, key, value, meta) => {
        const code = String(value ?? '');
        meta.state = meta.state || {};
        meta.state.mastercode = code;

        if (meta.logger && typeof meta.logger.info === 'function') {
            meta.logger.info(`YMZB-100: mastercode updated (${code.length} digits)`);
        }

        return {state: {mastercode: code}};
    },
};

const tzAssaLockWithMaster = {
    key: ['state'],
    convertSet: async (entity, key, value, meta) => {
        const desired = String(value).toUpperCase();
        const pin = meta?.state?.mastercode ?? '';

        const options = utils.getOptions(meta.mapped, entity);

        if (desired === 'LOCK') {
            await entity.command('closuresDoorLock', 'lockDoor',
                {pincodevalue: pin}, options);
            return {state: {state: 'LOCK'}};
        } else if (desired === 'UNLOCK') {
            await entity.command('closuresDoorLock', 'unlockDoor',
                {pincodevalue: pin}, options);
            return {state: {state: 'UNLOCK'}};
        } else {
            throw new Error(`Unsupported state value: ${value}`);
        }
    },
};

module.exports = [
    {
        fingerprint: [{
            profileID: 0x0104,
            deviceID: 0x000A,        // Door Lock
            endpointID: 1,
            inputClusters: [0x0101], // closuresDoorLock
        }],
        model: 'YMZB-100',
        vendor: 'Assa Abloy',
        description: 'Zigbee door lock module with master PIN auto injection',
        fromZigbee: [fz.lock],             
        toZigbee: [tzAssaLockWithMaster, tzAssaMastercode],
        exposes: [
            e.lock(),
            exposes.text('mastercode', exposes.access.STATE_SET)
                .withDescription('Master code used for lock/unlock'),
        ],

        configure: async (device, coordinatorEndpoint /*, logger */) => {
            const ep = device.getEndpoint(1);
            await reporting.bind(ep, coordinatorEndpoint, ['closuresDoorLock']);
        },
    },
];
```

**Comment by drtnnrs** [comment]
Any progress on this? Although the seller of this module (Homely in my case) are nice to share their API, they don't share write access on it, so I have to use their app to lock and unlock the door. Would love to pair it with Home Assistant so that I can enjoy more features and possibilites.

**Comment by danieledwardgeorgehitchcock** [comment]
This might have taken a backward, more than forward step, as the reverse engineering I did for the Doorman clusters will possibly be removed (at least commented out) as part of the wider piece of work to port manufacturer specific clusters into the manufacturer provider files instead of the generic zigbee provider. See: https://github.com/Koenkk/zigbee-herdsman/issues/1622#issuecomment-3953039266 and Koenkk/zigbee-herdsman-converters#11591

**Comment by tore1980** [comment]
Are there any updates on this?

**Reply by spacecakes** [reply]
Would be great to get the V2L locks working without middlemen. Verisure works reasonably OK with Home Assistant, but their API is slow and polling based
